scrapers/fireroad.py

```diff
@@ -26,34 +26,43 @@
 URL = "https://fireroad.mit.edu/courses/all?full=true"
 
 
-def parse_timeslot(day, slot):
-    """Parses a timeslot. Example: parse_timeslot("M", "10-11.30") -> [4, 3]
+def parse_timeslot(day, slot, pm):
+    """Parses a timeslot. Example: parse_timeslot("M", "10-11.30", False) -> [4, 3]
 
     Args:
     * day (str): The day as a string
     * slot (str): The slot as a string
+    * pm (bool): Whether the timeslot is in the evening
 
     Returns:
     * list[int]: The parsed day and timeslot
     """
-    pm, slot = slot.endswith(" PM"), slot.rstrip(" PM")
+    assert pm == slot.endswith(" PM")
```

Comment:
if this throws an error how is it handled?

Reply:
It gets logged to stdout and that class gets skipped, just like any other error in parsing a class's schedule:
https://github.com/sipb/hydrant/blob/e266b31e4566f3299714bc56fa3e329cebc7b5c6/scrapers/fireroad.py#L230-L235

(This is what happens to 15.089 on `main` and to the early morning classes on `timeslots`.)